src/CoreShop/Component/Core/Order/Processor/CartItemsProcessor.php

```diff
@@ -46,36 +52,48 @@ final class CartItemsProcessor implements CartProcessorInterface
      * @param PurchasableCalculatorInterface     $productPriceCalculator
      * @param QuantityReferenceDetectorInterface $quantityReferenceDetector
      * @param CartItemProcessorInterface         $cartItemProcessor
+     * @param CartContextResolverInterface       $cartContextResolver
      */
     public function __construct(
         PurchasableCalculatorInterface $productPriceCalculator,
         QuantityReferenceDetectorInterface $quantityReferenceDetector,
-        CartItemProcessorInterface $cartItemProcessor
+        CartItemProcessorInterface $cartItemProcessor,
+        CartContextResolverInterface $cartContextResolver = null
     ) {
         $this->productPriceCalculator = $productPriceCalculator;
         $this->quantityReferenceDetector = $quantityReferenceDetector;
         $this->cartItemProcessor = $cartItemProcessor;
+        $this->cartContextResolver = $cartContextResolver;
     }
 
     /**
      * {@inheritdoc}
      */
     public function process(CartInterface $cart)
     {
-        $store = $cart->getStore();
+        if (null === $this->cartContextResolver) {
+            @trigger_error(
+                'Using CartItemsProcessor without a CarContextResolverInterpreter is deprecated since 2.1.2 and will be removed with 3.0.0',
```

Comment:
```suggestion
                'Using CartItemsProcessor without a CartContextResolverInterface is deprecated since 2.1.2 and will be removed with 3.0.0',
```

Reply:
done